applications/DG-Max/Algorithms/DGMaxDiscretization.cpp

```diff
@@ -493,17 +493,28 @@ LinearAlgebra::SmallVector<DIM> DGMaxDiscretization<DIM>::computeField(
     const Base::Element* element, const Geometry::PointReference<DIM>& p,
     const LinearAlgebra::MiddleSizeVector& coefficients) const {
     logger.log(Log::WARN, "Only computing the real part of the field.");
+    return computeFields(element, p, coefficients).realEField;
+}
+
+template <std::size_t DIM>
+typename DGMaxDiscretization<DIM>::Fields
+    DGMaxDiscretization<DIM>::computeFields(
+        const Base::Element* element, const Geometry::PointReference<DIM>& p,
+        const LinearAlgebra::MiddleSizeVector& coefficients) const {
     Base::PhysicalElement<DIM> physicalElement;
     physicalElement.setElement(element);
     std::shared_ptr<Base::CoordinateTransformation<DIM>> transform{
         new Base::HCurlConformingTransformation<DIM>()};
     physicalElement.setTransformation(transform);
```

Comment:
The function setTransformation in `PhysicalElement` has as signature
```
void setTransformation(std::shared_ptr<Base::CoordinateTransformation<DIM>> transform, std::size_t unknown = 0)
```
I think this was done to because of the inheritance in the coordinate transformations. In the long run it might be good to build only one of these transformations and then share it everywhere.